### DISCUSSION TITLE
UDP messages send buffer size

### DISCUSSION THREAD

**Discussion by skaytex** [openthread/openthread#7464] [Q&A] [answered]
We are testing openthread stack and we want to know limitations and also we want to know how fast message going out from send buffer. We try to send many UDP messages in a row. **Is there any command to know that my message/s was out from buffer or is it possible to get this buffer size?** 

It's very important that this buffer not to overflow and not to lose packets that we sent. And also we want to control our message flow.

**Comment by jwhui** [comment, marked as answer]
Take a look at [OpenThread's message APIs](https://openthread.io/reference/group/api-message). Specifically, [otMessageBufferInfo()](https://openthread.io/reference/group/api-message#otmessagegetbufferinfo) provides informations of message buffers in use and where they are allocated.
